src/it/java/com/google/checkstyle/test/chapter2filebasic/rule21filename/FileNameTest.java

```diff
@@ -37,19 +33,22 @@ protected String getPackageLocation() {
     @Test
     public void testOuterTypeFilename1() throws Exception {
         final String filePath = getPath("InputFileName1.java");
-        verifyWithConfigParser(MODULES, filePath);
+        formatFileWithGoogleJavaFormat(filePath);
+        verifyWithWholeConfig(filePath);
     }
```

Comment:
Usage of the newly created method. The input file gets automatically formatted by google-java-format.

Reply:
@Zopsss , you don't need to migrate to whole config in same PR. In next PRs place don't do this.

Reply:
Can you explain why we are formatting the file with an external tool before we run our IT test?

This is overlaying the original input and causing a git difference after a run, correct? @romani this seems like it will violate our CI that there be no unstaged changes after running tests.

Reply:
@Zopsss , please reference issue with Google formatter usage.
and remove migration to Whole config from this PR, this becomes confusing for more people.

Reply:
It looks like issue is https://github.com/checkstyle/checkstyle/issues/15218 .

It still doesn't make sense why we are doing this.

We have told the creators of this tool that we follow google style and not them. (see https://github.com/checkstyle/checkstyle/issues/3899#issuecomment-310166261 , https://github.com/checkstyle/checkstyle/issues/3591#issuecomment-272269358 , https://github.com/checkstyle/checkstyle/issues/6527#issuecomment-470734957 ) Adding this tool only shows either we both follow google or one of us doesn't. If it is them, then they have a conflict they need to resolve. Are you expecting us to report issues to them? We have enough issues in our own repo. 

There has also been times where we aren't sure who is at fault because google style has an issue of being clear enough.

Reply:
I find it very confusing and unintuitive that we are formatting test inputs before running CheckStyle over them, @Zopsss can you point to where the explanation/reasoning is for this change?

Reply:
> I find it very confusing and unintuitive that we are formatting test inputs before running CheckStyle over them,

We're migrating the google tests from chapter wise testing to whole config testing, so all the input files needs to follow all the rules of google java guide. 

So we need to format input files according to that, for this @romani suggested to use google-java-format before verifying the input files, which will format the input file to follow google java guide rules. Though it only formats spacing, indentation stuff, it cannot fix variables names, grouping of overloaded constructors, etc. 

> @Zopsss can you point to where the explanation/reasoning is for this change?

most of the conversation happened on discord, I have pinged you there in the specific threads and you can also refer to this issue: https://github.com/checkstyle/checkstyle/issues/15218#issue-2398841662

Reply:
> most of the conversation happened on discord

This is a fail; all details for an issue should be in the issue description, not in a private chat room. Imagine you are some maintainer in the future, and are trying to understand why we are dynamically formatting files. You will not have access to discord (and shouldn’t have to), and therefore will not have the full context of why this change was done.

Please translate any relevant informations from discord into the issue.

Reply:
> It looks like issue is https://github.com/checkstyle/checkstyle/issues/15218 .

I have renamed the commit and PR as well to refer to issue 15218